projects/igniteui-angular/src/lib/directives/dragdrop/README.md

```diff
@@ -0,0 +1,103 @@
+# igxDrag
+**igxDrag** is a directive that enables dragging of elements around the page.
+
+## Usage
+```html
+<div igxDrag [hideBaseOnDrag]="true" [animateOnRelease]="true" *ngFor="let elem of draggableElems" class="draggableElem" (dragStart)="onDragStart()" (dragEnd)="onDragEnd()">
+    <span [style.margin]="'auto'">{{elem}}</span>
+</div>
+```
+
+## Getting Started
+
+### Introduction
+
+When an element inside your Angular application needs to be dragged from one place to another the `igxDrag` directive can be used to achieve this behavior. In combination with the `igxDrop` directive the placing of the dragged element can be done as well to have fully interactive application.
+
+### Basic configuration
+
+The `igxDrag` directive can be applied on any DOM element by just adding it to its template.
+
+```html
+<div igxDrag>Drag me</div>
+```
+
+### DOM Behavior
+
+By default when an element with `igxDrag` directive applied needs to be dragged the user needs to swipe at least 5px in any direction in order for the actual dragging to occur. Otherwise the interaction is considered as a click and the `dragClicked` event is emitted.
+
+When dragging occurs a drag ghost element is spawned that is positioned so it is always under the mouse cursor or when on touch where it is being touched. The original element is still present, but it can be hidden automatically when dragging starts with the `hideBaseOnDrag` input. 
+
+The dragging can be canceled by the user before it starts by setting the `cancel` property of the `dragStart` event to `true`. This will cancel the current dragging logic that could be used.
+
+After the user releases the mouse/touch the drag ghost element is removed from the DOM and if the `hideBaseOnDrag` is enabled it will make the original element visible again and the `dragEnd` event will be emitted. If the `animateOnRelease` input is set to `true` all this will execute after the default animation of the drag ghost is finished which consist of returning it from the last dragged position to the position of the original element. Then the drag ghost will be removed with the rest and the `returnMoveEnd` event will be emitted.
+
+## API
+
+### Inputs
+
+| Name | Type | Default Value | Description |
+| :--- | :--- | :--- | :--- |
+| igxDrag          | any | - | Input used to save data inside the `igxDrag` directive. This can be set when instancing `igxDrag` on an element. |
+| dragTolerance    | number | 5 | Indicates when the drag should start
+     * By default the drag starts after the draggable element is moved by 5px |
+| ghostImageClass  | string | '' | Sets a custom class that will be added to the `dragGhost` element. |
+| hideBaseOnDrag   | boolean | false | Sets if the draggable element should hide when when dragging starts. |
+| animateOnRelease | boolean | false | Enables/disables the draggable element animation when the element is released. |
+
+### Outputs
+
+| Name | Description | Cancelable |
```

Comment:
@skrustev Can you add "Argument Type" column as it is in the igChip Readme.md file.